verifier/verifier.go

```diff
@@ -249,37 +249,41 @@ func (v *Verifier) VerifyWithContext(ctx context.Context, c *x509.Certificate, o
 		res.Parents = parentsFromChains(res.ValidAtExpirationChains)
 	} else {
 		res.Parents = parentsFromChains(res.CurrentChains)
-	}
 
-	if opts.OneCRL != nil && opts.OneCRL.Check(c) != nil {
-		res.InRevocationSet = true
-	}
-	if !res.InRevocationSet && opts.CRLSet != nil {
-		for _, parent := range res.Parents {
-			if opts.CRLSet.Check(c, hex.EncodeToString(parent.SPKIFingerprint)) != nil {
-				res.InRevocationSet = true
-				break
+		//
+		// Check revocation on non expired cert only
+		//
+
+		if opts.OneCRL != nil && opts.OneCRL.Check(c) != nil {
```

Comment:
Should it be a flag if we check revocation of expired flags or not?

Reply:
no, if the cert is expired, it's already invalid and can be removed from CRL and not served by OCSP responders.